### ISSUE TITLE
As a Dev I want Vector3 To Have a CreateUp, CreateRight, and CreateForward 

### ISSUE THREAD

**Issue by AMZN-Gene** [o3de/o3de#10900]
**Is your feature request related to a problem? Please describe.**
It's often that I want to produce an world up vector, and instead of Vector3::CreateUp I have to remember that Z and use Vector3::CreateAxisZ

**Describe the solution you'd like**
Provide an Vector3::CreateUp (0,0,1). CreateDown (0,0,-1),
CreateRight(1,0,0), CreateLeft(-1,0,0), CreateForward(0,1,0), and CreateBack(0,-1,0)

**Comment by pollend** [comment]
is this consistent across the editor. I know sometimes Y is up and sometimes Z is up.

**Comment by AMZN-Gene** [comment]
_"sometimes Y is up and sometimes Z is up"_
That's a bit alarming; I would think we'd want consistency to avoid confusion.

**Comment by pollend** [comment]
> _"sometimes Y is up and sometimes Z is up"_ That's a bit alarming; I would think we'd want consistency to avoid confusion.

I'm not implying that the rest of the Editor does this; poor wording. just commenting on the fact that the definition of Up can be different. I like the explicit X, Y and Z

**Comment by amzn-pratikpa** [comment]
Tagging @o3de/sig-simulation for further discussion

**Comment by hultonha** [comment]
I don't think this is a bad idea and I wouldn't be averse to adding it (so long as there's a comment signalling that the current convention is for Z to be up in basically all cases such as World space and Model/Local space... maybe we should call it `CreateRhUp` for right handed? 🤔 Though that's a bit ugly. Tagging @greerdv for visibility too.

**Comment by AMZN-Gene** [comment]
I think that’s over thinking it. GlobalUp or WorldUp would also work, but simply seeing Up as a static method ppl would assume it’s the works up direction in the engine

**Comment by hultonha** [comment]
I think WorldUp/Left/Right sound good - to be honest for something this small I'd just add it and create a PR! 😅 I personally would be happy to approve and we can have sig/core and sig/simulation chime in too.

**Comment by lumberyard-employee-dm** [comment]
I think adding Up, Right and Forward is fine as that defines a right handed coordinate systems.
Adding Left would allow a left handed coordinate system be defined.
If @o3de/sig-simulation is good with the changes, they can set the priority on this issue(I am thinking priority minor?)

**Comment by hultonha** [comment]
I'm happy with these changes as a maintainer of sig/simulation (tagging chairs @lgleim and @epsilon-delta for visibility too).

The only potential problem is in some crazy situation, someone might decide to change the coordinate system to by Y-up (but that seems _very_ unlikely so it would be a shame to not add these helpers because of an unlikely potential conflict a long way down the line). That's my take anyway 😝

**Comment by lgleim** [comment]
I definitely see value in these static helper functions, given they come with very clear and concise comments, especially regarding the global reference coordinate system. 
Looking forward to a PR

**Comment by AMZN-Gene** [comment]
**Unity3d** Vector3.up
https://docs.unity3d.com/ScriptReference/Vector3-up.html

**Godot**  Vector3.UP
https://docs.godotengine.org/en/stable/classes/class_vector3.html

**CryEngine** Vector3.Up
https://docs.cryengine.com/pages/viewpage.action?pageId=29448783

**Comment by hultonha** [comment]
Please can you attend the next sig-simulation meeting @AMZN-Gene to discuss.

It is definitely more complicated than it seems. From a purely mathematical context having 'up' in the context of a vector actually doesn't make sense (I'm sure @greerdv would have an opinion about this too 😉). Just because the other engines have this doesn't mean it's right. At the end of the call we were leaning towards free functions `GetWorldUp` etc that aren't enclosed with `Vector3`, this might be a reasonable compromise.

**Comment by AMZN-Gene** [comment]
It's in Vec3 classes for convenience. You're creating a new vector so it's natural to find it there. A free-floating-function is good too, but I'd suspect documentation would become more necessary to learn about it. 

I think everyone agrees that in 90% of o3de games you'd find a Vector3(1,0,0) Vector3(0,1,0), and Vector3(0,0,1) hardcoded somewhere; now it's just up to "where do I put this?"

**Comment by hultonha** [comment]
I mean we do have GetAxisX(), GetAxisY() and GetAxisZ() which are those values hard-coded. I just do think it's tricky associating a 'space' concept with a Vector3... I know it's very particular and fussy but I do think adding it to Vector3 is probably wrong... Free functions on the other hand with slightly clearer naming I thing would be a great addition

**Comment by greerdv** [comment]
I'd also prefer it not to live inside Vector3, since it's a higher level concept. On the other hand, there are already a few places in the low level maths libraries where our global conventions creep in e.g. Transform::CreateLookAt.

One other option I'd suggest considering is using behavior constants instead of Get... functions.

**Comment by hultonha** [comment]
Discussed again in sig-simulation triage. We agreed we would like to see this in some form, just not as static member functions. Having this as constants defined somewhere seems like the best plan. Please open a PR with these changes if you'd like to see this sooner rather than later. Thanks!
